nav2_toolkit/src/pose_saver_node.cpp

```diff
@@ -0,0 +1,296 @@
+#include <chrono>
+#include <fstream>
+#include <memory>
+#include <cstdio>
+#include <rclcpp/rclcpp.hpp>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <std_srvs/srv/trigger.hpp>
+#include <yaml-cpp/yaml.h>
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include "nav2_toolkit/pose_saver_node.hpp"
+
+using namespace std::chrono_literals;
+
+namespace nav2_toolkit
+{
+
+PoseSaverNode::PoseSaverNode()
+: Node("pose_saver_node"), saving_active_(false)
+{
+  this->declare_parameter<double>("save_interval_sec", 5.0);
+  this->declare_parameter<std::string>(
+    "pose_file_path", 
+    ament_index_cpp::get_package_share_directory("nav2_toolkit") + "/config/last_known_pose.yaml");
+  
+  this->declare_parameter<bool>("auto_start_saving", true);
+  this->declare_parameter<bool>("auto_restore_pose", true);
+
+  pose_file_path_ = this->get_parameter("pose_file_path").as_string();
+  double interval_sec = this->get_parameter("save_interval_sec").as_double();
+  bool auto_start = this->get_parameter("auto_start_saving").as_bool();
+  auto_restore_ = this->get_parameter("auto_restore_pose").as_bool();
+
+  timer_ = this->create_wall_timer(
+    std::chrono::duration<double>(interval_sec),
+    std::bind(&PoseSaverNode::timer_callback, this));
+  timer_->cancel();  // Disabled until service starts
+
+  sub_ = this->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+    "/amcl_pose", 10,
```

Comment:
For all topics in pub/sub/services: remove the `/` which sets it to be the global namespace. That doesn't allow this to work with namespaced systems